nixos/modules/services/development/livebook.nix

```diff
@@ -0,0 +1,101 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.services.livebook;
+in
+{
+  options.services.livebook = {
+    enableUserService = mkOption {
+      type = types.bool;
+      default = false;
+      description = lib.mdDoc ''
+        Whether to enable a user service for Livebook.  This implies
+        `installUserService`.
+      '';
+    };
+
+    installUserService = mkOption {
+      type = types.bool;
+      default = false;
+      description = lib.mdDoc ''
+        Whether to install a user service for Livebook.  Unless
+        `enableUserService` is also set, users will have to manually activate
+        the service with `systemctl --user enable livebook`.
+      '';
+    };
+
+    environment = mkOption {
+      type = with types; attrsOf str;
+      description = lib.mdDoc ''
+        The environment to pass to the Livebook service.  You need to set at least
+        `LIVEBOOK_PASSWORD`.  See `livebook server --help` for other options.
+      '';
+      example = {
+        LIVEBOOK_PASSWORD = "a password at least 30 characters long";
+        LIVEBOOK_PORT = 30123;
+      };
+    };
+
+    erlang_node_short_name = mkOption {
+      type = with types; nullOr str;
+      default = null;
+      example = "livebook";
+      description = "A short name for the distributed node.";
+    };
+
+    erlang_node_name = mkOption {
+      type = with types; nullOr str;
+      default = null;
+      example = "livebook@127.0.0.1";
+      description = "The name for the app distributed node.";
+    };
+
+    port = mkOption {
+      type = with types; nullOr port;
+      default = null;
+      example = 30123;
+      description = "The port to start the web application on.";
+    };
+
+    cookie = mkOption {
+      type = with types; nullOr str;
+      default = null;
+      description = "The cookie for the app distributed node.";
+    };
+
+    address = mkOption {
+      type = with types; nullOr str;
+      default = null;
+      description = lib.mdDoc ''
+        The address to start the web application on.  Must be a valid IPv4 or
+        IPv6 address.
+      '';
+    };
+  };
+
+  config = mkIf (cfg.installUserService || cfg.enableUserService) {
+    systemd.user.services.livebook = {
+      environment = cfg.environment;
+      serviceConfig = {
+        Restart = "always";
+        ExecStart = ''\
+          ${pkgs.livebook}/bin/livebook server \
+          ${optionalString (cfg.erlang_node_short_name != null)
+            "--sname ${cfg.erlang_node_short_name}"} \
+          ${optionalString (cfg.erlang_node_name != null)
+            "--name ${cfg.erlang_node_name}"} \
+          ${optionalString (cfg.port != null)
+            "--port ${toString cfg.port}"} \
+          ${optionalString (cfg.cookie != null)
+            "--cookie ${cfg.cookie}"} \
+          ${optionalString (cfg.address != null)
+            "--ip ${cfg.address}"} \
+        '';
+      };
+      path = [ pkgs.bash ];
```

Comment:
I thought this was already guaranteed. Did you hit an error, causing you to add this?

Reply:
Just saw your comment in the PR title. I'd love to know what's going on here. If anyone knows, please leave a comment!

Reply:
I have no idea what causes the error in Erlang and it doesn't have any useful details.  When I first hit it, I searched for hours and I couldn't find anyone else who's run into this (presumably because everybody has bash).

Reply:
I found the offending code in Erlang.  One of the monitors calls `df`: https://github.com/erlang/otp/blob/95de10042b9492f0414793e9217711c5a4d19504/lib/os_mon/src/disksup.erl#L288

Specifically, it calls it with an invocation like: `(/bin/df -lk -x squashfs PATH \n) </dev/null; echo  "^M"\n`.

My Erlang's very rusty, but I think this then runs in a `sh` shell: https://github.com/erlang/otp/blob/95de10042b9492f0414793e9217711c5a4d19504/lib/os_mon/src/disksup.erl#L242C1-L242C1

So, I guess what we really need is for a call to `sh` (without any pat) to work.  Adding `bash` to the environment achieves this.

Reply:
Does this mean that the erlang package should be amended to always include bash?

Reply:
I see the RabbitMQ and ejabberd services both depend on `coreutils` instead, although that might be because they have to find and create data directories.

I think it's accurate to say that "Erlang services need at least a working shell (and maybe other things in order to run)".  Bash happens to be sufficient for Livebook.  I have not tried other shells, so I'm not really confided in saying it always needs bash.

Reply:
From the pleroma service:

```
# disksup requires bash
path = [ pkgs.bash ];
```

I'm going through all the Elixir apps now and I'll post what I find.

Reply:
The akkoma service adds coreutils to the path (and a bunch of other things).

Reply:
The mobilizon service adds gawk, imagemagick, libwebp, file, gifsicle, jpegoptim, optipng, pngquant.  I don't know where it's getting `sh` from.

Reply:
That's all the BEAM services.  So, two of them add `bash` to the path, three add `coreutils`, and one adds a bunch of other stuff.

If I had to guess what's going on, different apps in the Erlang distribution need different things (e.g. maybe `mnesia` needs `coreutils` because it does stuff with directories and files).  Or maybe the shell scripts that the Erlang release process generates need different combinations of programs.

I don't know what the general thing to do here is.